src/core/etl/src/Flow/ETL/Join/Comparison/All.php

```diff
@@ -18,9 +19,13 @@ final class All implements Comparison
      */
     private array $comparisons;
 
-    public function __construct(Comparison $comparison, Comparison ...$comparisons)
+    public function __construct(Comparison ...$comparisons)
```

Comment:
I prefer to keep it like it is now. This way, the constructor enforces at least one Comparison, so there is no need to write a unit test covering this edge case to satisfy mutation tests. 
It's an approach commonly used in Scala when at least one element is required. 
This particular array_merge should also have no impact on the pipeline processing performance.